### PR TITLE
Swagger Completeness for Microsoft.Authorization

### PR DESCRIPTION
**Pull request by tjegbejimba**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/locks.json [hunk 1]
```diff
@@ -0,0 +1,1144 @@
...
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed."
+          }
```
Comment (j5lim) [inline specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/locks.json:58]: Need to add schema for error. You can reference common-types https://github.com/Azure/azure-rest-api-specs/blob/002ad52075e725708deabf842a736c98751b8566/specification/common-types/resource-management/v2/types.json#L322

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/examples/ManagementLocks_CreateOrUpdateAtResourceGroupLevel.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      }
+    }
+  },
+  "responses": {
+    "201": {
+      "body": {
+        "properties": {
+          "level": "ReadOnly"
+        },
+        "id": "/providers/Microsoft.Authorization/locks/testlock",
```
Comment (j5lim) [inline specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/examples/ManagementLocks_CreateOrUpdateAtResourceGroupLevel.json:19]: ID should be based on the request scope (i.e. /subscriptions/subscriptionId/resourceGroups/resourcegroupname/providers/Microsoft.Authorization/locks/testlock). I see id is not correct in all examples. Better to fix them.

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2017-04-01/locks.json [hunk 3]
```diff
@@ -0,0 +1,1089 @@
...
+          "Operations"
+        ],
+        "operationId": "AuthorizationOperations_List",
+        "description": "Lists all of the available Microsoft.Authorization REST API operations.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```
Comment (j5lim) [inline specification/resources/resource-manager/Microsoft.Authorization/stable/2017-04-01/locks.json:49]: Need default response to handle error.
https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review

**Review by j5lim — CHANGES_REQUESTED** [review]

**Review by j5lim — APPROVED** [review]
Approved from ARM side.

**Comment by raych1** [comment]
@jianyexi , can you have a look at the breaking change check runtime errors?

**Comment by jianyexi** [comment]
The files in the folder [stable/2020-05-01](https://github.com/Azure/azure-rest-api-specs/tree/6d60a192d30f22be20e046a220e8e812672038d8/specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01) refer to two versions of common types.json , which caused the conflict of 'ErrorReponse' ,  it might be fixed by unifying the reference

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/locks.json [hunk 4]
```diff
@@ -0,0 +1,1188 @@
...
+    "/providers/Microsoft.Authorization/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "AuthorizationOperations_List",
+        "description": "Lists all of the available Microsoft.Authorization REST API operations.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ApiVersionParameter"
```
Comment (jianyexi) [inline specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/locks.json:46]:
```suggestion
            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
```

**Comment by raych1** [comment]
@ArcturusZhang , @msyyc , can you approve on the breaking change for GO/Python SDK?

**Review by jianyexi — APPROVED** [review]

**Comment by msyyc** [comment]
approve for python

**Pull request merged**